packages/backend-defaults/src/entrypoints/rootHttpRouter/http/MiddlewareFactory.ts

```diff
@@ -227,6 +231,85 @@ export class MiddlewareFactory {
     return cors(readCorsOptions(this.#config.getOptionalConfig('backend')));
   }
 
+  /**
+   * Returns a middleware that implements rate limiting.
+   *
+   * @remarks
+   *
+   * Rate limiting is a common technique to prevent abuse of APIs. This middleware is
+   * configured using the config key `backend.rateLimit`.
+   *
+   * @returns An Express request handler
+   */
+  rateLimit(): RequestHandler {
+    const enabled = this.#config.has('backend.rateLimit');
+    if (!enabled) {
+      return (_req: Request, _res: Response, next: NextFunction) => {
+        next();
+      };
+    }
+
+    const useDefaults = this.#config.getOptional('backend.rateLimit') === true;
+    const rateLimitOptions = useDefaults
+      ? undefined
+      : this.#config.getOptionalConfig('backend.rateLimit');
+
+    let windowMs: number = 60000;
+    if (rateLimitOptions && rateLimitOptions.has('window')) {
+      const windowDuration = readDurationFromConfig(rateLimitOptions, {
+        key: 'window',
+      });
+      windowMs = durationToMilliseconds(windowDuration);
+    }
+
+    const ipAllowList = rateLimitOptions?.getOptionalStringArray(
+      'ipAllowList',
+    ) ?? ['127.0.0.1', '0:0:0:0:0:0:0:1', '::1'];
+
+    return rateLimit({
+      windowMs,
+      limit: rateLimitOptions?.getOptionalNumber('incomingRequestLimit'),
+      skipSuccessfulRequests: rateLimitOptions?.getOptionalBoolean(
+        'skipSuccessfulRequests',
+      ),
+      message: {
+        error: {
+          name: 'Error',
+          message: `Too many requests, please try again later`,
+        },
+        response: {
+          statusCode: 429,
+        },
+      },
+      statusCode: 429,
+      skipFailedRequests:
+        rateLimitOptions?.getOptionalBoolean('skipFailedRequests'),
+      passOnStoreError:
+        rateLimitOptions?.getOptionalBoolean('passOnStoreError'),
+      keyGenerator(req, _res): string {
+        if (!req.ip) {
+          return req.socket.remoteAddress!;
+        }
+        return req.ip;
+      },
+      skip: (req, _res) => {
+        return (
+          Boolean(req.ip && ipAllowList.includes(req.ip)) ||
+          Boolean(
+            req.socket.remoteAddress &&
+              ipAllowList.includes(req.socket.remoteAddress),
+          )
+        );
+      },
+      validate: {
+        trustProxy: false,
```

Comment:
should this be false always?

Reply:
Yep, the app setting controls this here: https://github.com/backstage/backstage/pull/28708/files/a70c2429dbf419130f9ea158455473947102760b#diff-86ad1b6a694dd250823aee39d410428dd837c9d9a04ca8c33bd1081fbe3f22afR120